src/Adapter/Driver/IbmDb2/Connection.php

```diff
@@ -149,7 +149,28 @@ public function isConnected()
     public function disconnect()
     {
         if ($this->resource) {
-            db2_close($this->resource);
+            
+            // localize
+            $p = $this->connectionParameters;
+
+            // given a list of key names, test for existence in $p
+            $findParameterValue = function (array $names) use ($p) {
```

Comment:
Make a new private method returning `bool` instead. Also, `array_change_key_case()` is probably a better approach, since nothing guarantees that mIxEDCasE won't be given as parameter.